cypress/e2e/trafficlight/register.spec.ts

```diff
@@ -0,0 +1,106 @@
+                cy.request('POST', respondUrl, { response: "registered" }).then((response) => {
+                    expect(response.status).to.eq(200);
+                });
+                break;
+            case "login":
+                cy.visit("http://localhost:8083/#/login");
+                cy.get("#mx_LoginForm_username", { timeout: 15000 }).should("be.visible");
+                cy.get(".mx_ServerPicker_change").click();
+                cy.get(".mx_ServerPickerDialog_otherHomeserver").type(data['homeserver_url']['local']);
+                cy.get(".mx_ServerPickerDialog_continue").click();
+                // wait for the dialog to go away
+                cy.get('.mx_ServerPickerDialog').should('not.exist');
+                cy.get("#mx_LoginForm_username").type(data['username']);
+                cy.get("#mx_LoginForm_password").type(data['password']);
+                cy.get(".mx_Login_submit").click();
+                cy.request('POST', respondUrl, { response: "loggedin" }).then((response) => {
+                    expect(response.status).to.eq(200);
+                });
+                break;
+            case "start_crosssign":
+                cy.get('.mx_CompleteSecurity_actionRow > .mx_AccessibleButton').click();
+                cy.request('POST', respondUrl, { response: "started_crosssign" }).then((response) => {
+                    expect(response.status).to.eq(200);
+                });
+                break;
+            case "accept_crosssign":
+                // Can we please tag some buttons :)
+                // Click "Verify" when it comes up
+                cy.get('.mx_AccessibleButton_kind_primary').click();
+                // Click to move to emoji verification
+                cy.get('.mx_VerificationPanel_QRPhase_startOption > .mx_AccessibleButton').click();
+                cy.request('POST', respondUrl, { response: "accepted_crosssign" }).then((response) => {
+                    expect(response.status).to.eq(200);
+                });
+                break;
+            case "verify_crosssign_emoji":
+                cy.get('.mx_VerificationShowSas_buttonRow > .mx_AccessibleButton_kind_primary').click();
+                cy.get('.mx_UserInfo_container > .mx_AccessibleButton').click();
+                cy.request('POST', respondUrl, { response: "verified_crosssign" }).then((response) => {
+                    expect(response.status).to.eq(200);
+                });
+                break;
+            case "idle":
+                cy.wait(5000);
+                break;
+            case "exit":
+                cy.log("Client asked to exit, test complete or server teardown");
+                return;
+            default:
+                cy.log("WARNING: unknown action ", action);
+                return;
+        }
+        recurse();
+    });
+}
+
+describe("traffic light cycle", () => {
+    it("run trafficlight client", () => {
```

Comment:
Tests should be phrased in a way they make grammatical sense, `it run trafficlight client` doesn't quite accomplish that

Reply:
I believe i have adjusted this.